### PR TITLE
Use cf-xarray

### PR DESCRIPTION
**Pull request by aulemahal**
This modifies the way coordinates and bounds are read in the frontend to allow the use of cf-xarray and relax naming rules.

1D Bounds can be computed with cf-xarray, allowing the use of 'conservative' methods without passing 'lon_b' and 'lat_b' explicitly. However, the specific method that converts CF-bounds ( shape (N, 2) or (N, M, 4)) to xESMF-bounds is not yet in the released cf-xarray. So this PR will have to wait.

Also, I started from #24, as some of code from this branch needed modification too.

This still needs some documentation (maybe a notebook example).

### CODE REVIEW

**Comment by dcherian** [comment]
Exciting! We can do a cf_xarray release to make this easy to use

File: xesmf/frontend.py [hunk 1]
```diff
@@ -514,15 +554,21 @@ def __init__(
...
-            by 0, and non-masked values by 1.
+            Contain input and output grid coordinates.
+            All variables that the cf-xarray accessor understand are accepted.
+            Otherwise, look for ``lon``, ``lat``,
+            optionally ``lon_b``, ``lat_b`` for conservative methods,
+            and ``mask``. Note that for `mask`, the ESMF convention is used,
+            where masked values are identified by 0, and non-masked values by 1.
+
+            For conservative methods, if bounds are not present, they will be
+            computed using `cf-xarray` (currently only valid for 1D coordinates).
```
Comment (huard) [inline xesmf/frontend.py:565]:
```suggestion
            computed using `cf-xarray` (only 1D coordinates are currently supported).
```

File: xesmf/frontend.py [hunk 2]
```diff
@@ -756,16 +802,22 @@ def __init__(
...
-            data_vars={
-                'lon': (('poly',), [c[0][0] for c in poly_centers]),
-                'lat': (('poly',), [c[1][0] for c in poly_centers]),
-            }
-        )
+        self._lon_out = np.asarray([c[0][0] for c in poly_centers])
+        self._lat_out = np.asarray([c[1][0] for c in poly_centers])
+
+        # We put names 'lon' and 'lat' so ds_to_ESMFlocstream finds them easily.
+        # _lon_out_name and _lat_out_name are used on the ouput anyway.
```
Comment (huard) [inline xesmf/frontend.py:818]:
```suggestion
        # _lon_out_name and _lat_out_name are used on the output anyway.
```

**Review by huard — APPROVED** [review]

**Review by stefraynaud — APPROVED** [review]
Excellent!

**Comment by huard** [comment]
Thanks @dcherian for the release !

**Pull request merged**